zebra-consensus/src/block/subsidy/general.rs

```diff
@@ -0,0 +1,329 @@
+//! Block and Miner subsidies, halvings and target spacing modifiers. - [§7.7][7.7]
+//!
+//! [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
+
+use std::convert::TryFrom;
+
+use zebra_chain::{
+    amount::{Amount, Error, NonNegative},
+    block::Height,
+    parameters::{Network, NetworkUpgrade::*},
+    transaction::Transaction,
+    transparent,
+};
+
+use crate::parameters::subsidy::*;
+
+/// The divisor used for halvings.
+///
+/// `1 << Halving(height)`, as described in [protocol specification §7.7][7.7]
+///
+/// [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
+pub fn halving_divisor(height: Height, network: Network) -> u64 {
+    let blossom_height = Blossom
+        .activation_height(network)
+        .expect("blossom activation height should be available");
+
+    match height {
+        height if (height >= SLOW_START_SHIFT && height < blossom_height) => {
+            let scaled_pre_blossom_height = (height - SLOW_START_SHIFT) as u64;
+            let halving_shift = scaled_pre_blossom_height / (PRE_BLOSSOM_HALVING_INTERVAL.0 as u64);
+            1 << halving_shift
+        }
+        height if (height >= blossom_height) => {
+            let scaled_pre_blossom_height =
+                (blossom_height - SLOW_START_SHIFT) as u64 * BLOSSOM_POW_TARGET_SPACING_RATIO;
+            let post_blossom_height = (height - blossom_height) as u64;
+            let halving_shift = (scaled_pre_blossom_height + post_blossom_height)
+                / (POST_BLOSSOM_HALVING_INTERVAL.0 as u64);
+            1 << halving_shift
+        }
+        _ => unreachable!(
+            "unsupported block height: callers should handle blocks below {:?}",
+            SLOW_START_SHIFT
+        ),
+    }
```

Comment:
same as above, prefer if statements over match when not relying on pattern matching

Reply:
I think that's a good change here, I'll push it after I commit your suggestions.

Reply:
Done in https://github.com/oxarbitrage/zebra/pull/30